### PR TITLE
17 review new users api and update vehicle api

### LINKED ISSUE #17: Review New Users API and update Vehicle API
Recently a new user api was added. Study that code and edit your existing vehicle api accordingly

### PR DESCRIPTION
**Pull request by Charhmwang**
Implemented getAllVehicles & getVehicleById methods.
Still in progress of updating other methods.
Please ignore #23 PR request

### CODE REVIEW

File: src/main/java/com/jlr/ttl/ds/api/controllers/VehicleController.java [hunk 1]
```diff
@@ -12,69 +15,55 @@
...
-    private VehicleRepository vehicleRepository;
 
     @GetMapping("/all")
-    public ResponseEntity<List<Vehicle>> getAllVehicles() {
-        List<Vehicle> response = vehicleService.getAllVehiclesService();
-        return new ResponseEntity<>(response, HttpStatus.OK);
-    }
+    public ResponseEntity<DSResponse<List<VehicleResponse>>> getAllVehicles() {
+        List<VehicleResponse> dsVehicleResponse = null;
+        try{
```
Comment (BFARIAAL) [inline src/main/java/com/jlr/ttl/ds/api/controllers/VehicleController.java]: Remove empty line

File: src/main/java/com/jlr/ttl/ds/api/controllers/VehicleController.java [hunk 2]
```diff
@@ -12,69 +15,55 @@
...
+                .status(DSConstants.STATUS_SUCCESS)
+                .results(dsVehicleResponse)
+                .build(), HttpStatus.OK);
     }
 
-    @GetMapping("/by-loc/{loc_code}")
-    public ResponseEntity<?> findVehiclesByLoc(@PathVariable(value = "loc_code") String loc_code) {
+    @GetMapping("/by_id/{id}")
+    public ResponseEntity<DSResponse<VehicleResponse>> getVehicleById(@PathVariable(value = "id") String id){
+
```
Comment (BFARIAAL) [inline src/main/java/com/jlr/ttl/ds/api/controllers/VehicleController.java]: Remove empty line

File: src/main/java/com/jlr/ttl/ds/api/controllers/VehicleController.java [hunk 3]
```diff
@@ -12,69 +15,55 @@
...
+        VehicleResponse vehicleResponse = null;
 
         try {
-            return ResponseEntity.ok(vehicleService.findVehiclesByLocService(loc_code));
-        } catch (IllegalStateException e) {
-            return ResponseEntity.ok("Illegal location!");
+            vehicleResponse = vehicleService.getVehicleByID(id);
+        } catch (IllegalStateException illegalStateException) {
```
Comment (BFARIAAL) [inline src/main/java/com/jlr/ttl/ds/api/controllers/VehicleController.java]: Do you expect to catch an IllegalStateException?

File: src/main/java/com/jlr/ttl/ds/api/dto/request/VehicleRequest.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+@AllArgsConstructor
+@NoArgsConstructor
+@EqualsAndHashCode
+public class VehicleRequest implements DSRequestInterface<Vehicle> {
+
+    private String id;
+    private String vin;
+    private String locCode;
+    private Date endDate;
+
```
Comment (BFARIAAL) [inline src/main/java/com/jlr/ttl/ds/api/dto/request/VehicleRequest.java]: Remove empty lines

File: src/main/java/com/jlr/ttl/ds/api/dto/response/VehicleResponse.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@EqualsAndHashCode
+public class VehicleResponse implements DSResponseInterface<Vehicle> {
+    private String id;
+    private String vin;
+    private String locCode;
+    private Date endDate;
+
```
Comment (BFARIAAL) [inline src/main/java/com/jlr/ttl/ds/api/dto/response/VehicleResponse.java]: Remove empty lines

File: src/main/java/com/jlr/ttl/ds/api/repositories/VehicleRepository.java [hunk 6]
```diff
@@ -2,15 +2,14 @@
...
 
 import java.util.List;
-@Repository
-public interface VehicleRepository
-        extends JpaRepository<Vehicle, String> {
-    @Query("SELECT v FROM Vehicle v WHERE v.loc = :loc")
-    List<Vehicle> findVehiclesByLoc(String loc);
+
+public interface VehicleRepository extends JpaRepository<VehiclesTable, String> {
+//    @Query("SELECT v FROM Vehicle v WHERE v.loc = :loc")
```
Comment (BFARIAAL) [inline src/main/java/com/jlr/ttl/ds/api/repositories/VehicleRepository.java]: Do not push commented code. Please remove it

File: src/main/java/com/jlr/ttl/ds/api/services/VehicleService.java [hunk 7]
```diff
@@ -1,63 +1,72 @@
...
+     * Getting all the vehicles from database.
+     * @return a list of VehicleResponse
+     */
+    public List<VehicleResponse> getAllVehicles() throws ServiceBusinessException {
+        try {
+            List<VehiclesTable> dbResponse = vehicleRepository.findAll();
 
-        existingVehicle.setLoc_code(vehicle.getLoc_code());
-        vehicleRepository.save(existingVehicle);
+            List<VehicleResponse> finalList = new ArrayList<VehicleResponse>();
```
Comment (BFARIAAL) [inline src/main/java/com/jlr/ttl/ds/api/services/VehicleService.java]: Refractor this variable to have a more descriptive name such as listOfVehicleResponse

**Review by BFARIAAL — CHANGES_REQUESTED** [review]
Really great job with this PR. You were able to successfully study the DSUsers api and implement a similar logic here. The comments left are mostly for code cleaning.
Again, congratulations for the good work.

File: src/main/java/com/jlr/ttl/ds/api/controllers/VehicleController.java (hunk 3 above)
Comment (Charhmwang) [reply src/main/java/com/jlr/ttl/ds/api/controllers/VehicleController.java]: I don't understand this part. And I didn't see line 44 45 code from my last commit. Could you please explain a bit more?

Comment (BFARIAAL) [reply src/main/java/com/jlr/ttl/ds/api/controllers/VehicleController.java]: Your service layer is not throwing an IllegalStateException so I'm not sure why we are specifically catching this type of error. Can't this be handled with a generic catch(Exception ex)? Are you aware of any specific issues

**Comment by Charhmwang** [comment]
> Really great job with this PR. You were able to successfully study the DSUsers api and implement a similar logic here. The comments left are mostly for code cleaning. Again, congratulations for the good work.

Thank you so much for this positive feedback! I'd add more features for Vehicle route and meanwhile starting resolve the new ticket I'm assigned. Will also bear in mind to keep the coding convention aligning to the User ones you provided :)

File: src/main/java/com/jlr/ttl/ds/api/services/VehicleService.java [hunk 8]
```diff
@@ -69,4 +70,35 @@ public List<VehicleResponse> getAllVehicles() throws ServiceBusinessException {
         }
     }
 
+    @TrackExecutionTime
```
Comment (BFARIAAL) [inline src/main/java/com/jlr/ttl/ds/api/services/VehicleService.java]: Remove the code past this line

File: src/main/java/com/jlr/ttl/ds/api/services/VehicleService.java [hunk 9]
```diff
@@ -54,6 +54,7 @@ public VehicleResponse getVehicleByID(String id) throws ServiceBusinessException
      * Getting all the vehicles from database.
      * @return a list of VehicleResponse
      */
+    @TrackExecutionTime
```
Comment (BFARIAAL) [inline src/main/java/com/jlr/ttl/ds/api/services/VehicleService.java]: Remove Annotation

**Review by BFARIAAL — CHANGES_REQUESTED** [review]
Please resolve the last three comments

File: src/main/java/com/jlr/ttl/ds/api/controllers/VehicleController.java (hunk 3 above)
Comment (Charhmwang) [reply src/main/java/com/jlr/ttl/ds/api/controllers/VehicleController.java]: I replaced the ServiceBusinessException thrown in the VehicleService into IllegalStateException. Is that okay?

File: src/main/java/com/jlr/ttl/ds/api/services/VehicleService.java [hunk 10]
```diff
@@ -46,59 +46,26 @@ public VehicleResponse getVehicleByID(String id) throws ServiceBusinessException
             log.warn(errorMessage);
             throw new VehicleNotFoundException(errorMessage);
         }catch (Exception ex) {
-            throw new ServiceBusinessException("Exception occurred while fetching vehicle with id : " + id);
+            throw new IllegalStateException("Exception occurred while fetching vehicle with id : " + id);
```
Comment (BFARIAAL) [inline src/main/java/com/jlr/ttl/ds/api/services/VehicleService.java]: Revert this change. Change the exception in the controller instead

File: src/main/java/com/jlr/ttl/ds/api/controllers/VehicleController.java (hunk 1 above)
Comment (BFARIAAL) [reply src/main/java/com/jlr/ttl/ds/api/controllers/VehicleController.java]: You added this empty line again

**Review by BFARIAAL — CHANGES_REQUESTED** [review]

**Review by BFARIAAL — APPROVED** [review]
PR ready to be merged. Good job
